### PR TITLE
Mainline lamellar armor from Medieval Stuff mod

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

```SUMMARY: Content "Mainline lamellar armor from Medieval Stuff mod"```
#### Purpose of change

Game already has medieval items in mainline, so could have this one as well. Also prevents the mod from becoming broken or bugged at some point because of mainline changes.
#### Describe the solution

Move item entry, recipe entry, and item item group entries from mod JSON files to core JSON files. Also updates No Medieval Items blacklist mod to include newly-moved item as well.
#### Describe alternatives you've considered

Changing recipe to require sewing skill only (likely at a higher level). It involves working with leather and cloth, after all, which is what sewing usually does.
#### Additional context

None.

### CODE REVIEW

File: data/json/recipes/armor/suit.json [hunk 1]
```diff
@@ -100,6 +100,23 @@
...
+  {
+    "result": "armor_lamellar",
+    "type": "recipe",
+    "category": "CC_ARMOR",
+    "subcategory": "CSC_ARMOR_SUIT",
+    "skill_used": "fabrication",
+    "difficulty": 4,
+    "skills_required": [ "tailor", 3 ],
+    "time": 30000,
+    "book_learn": [ [ "textbook_tailor", 5 ], [ "tailor_portfolio", 5 ], [ "textbook_armschina", 4 ] ],
```
Comment (I-am-Erk) [inline data/json/recipes/armor/suit.json]: I'd say that lamellar armour probably wouldn't be found in a tailoring textbook or portfolio. Arms machina is ok. On the other hand, it's actually a pretty straightforward thing to make if you have a guide. The minimum level to learn the recipe could be lower.

**Comment by Photoloss** [comment]
> I like lamellar armour, but I'm not sure it needs to be a different recipe from leather armour or an o-yoroi

Lamellar could be a low-skill-high-effort craft. As you say it's pretty straightforward so maybe even an autolearn recipe at high survival+tailoring for a more realistic innawoods/illiterate option. Equal protection but slightly higher encumbrance compared to following the book.

Of course doing that justice requires a *vast* increase in crafting time, likely #28775 or an intermediate "sheet of leather scales" item.

If instead following a proper reference lamellar armour would have slightly lower encumbrance but also slightly lower protection as the individual plates make it more flexible but leave certain angles exposed.

**Comment by I-am-Erk** [comment]
Yeah, when ifreund finishes making crafting-by-percentage there may be a bit more role for lamellar as an easier, slower armour that uses smaller bits. I'm just not convinced it's sufficiently different to need an entire item definition, rather than just being an alternate (easier+slower) recipe for leather armour.

**Comment by Photoloss** [comment]
> 
> 
> Yeah, when ifreund finishes making crafting-by-percentage there may be a bit more role for lamellar as an easier, slower armour that uses smaller bits. I'm just not convinced it's sufficiently different to need an entire item definition, rather than just being an alternate (easier+slower) recipe for leather armour.

Depends on how anal Kevin is about coverage, a perfect 100% protection could be reasonable with professionally crafted gambeson/bulk leather but there would always be gaps in between the plates/scales in lamellar. Not that zombies or beasts could actually exploit those gaps with their oversized anatomy.

**Comment by Slumbrero** [comment]
> If instead following a proper reference lamellar armour would have slightly lower encumbrance but also slightly lower protection as the individual plates make it more flexible but leave certain angles exposed.

Yes, encumbrance should be lower as well as bash protection mainly, cut protection should be almost the same.

**Comment by kevingranade** [comment]
I'm not sure these are that similar at all, most lamellar leather examples I see are cuirasses built of hard leather plates, which reinforces the "easy to make" feature, whereas the leather armor item seems to be oriented as a more complete suit (not that I'm sure that's at all accurate...).

If you're going for a easy to craft historically based lamellar cuirass, 80% torso and no leg or arm coverage seems appropriate rather than a lamellar suit similar to an O-yori.

**Comment by Rail-Runner** [comment]
Taken into account all the suggestions. It's now easy to make and takes lots of time, and only protects torso.

**Pull request merged**